main.tf

```diff
@@ -1,9 +1,36 @@
 resource "azurerm_databricks_workspace" "this" {
-  name                        = var.workspace_name
-  resource_group_name         = var.resource_group_name
-  location                    = var.location
-  sku                         = var.sku
-  managed_resource_group_name = var.managed_resource_group_name
+  name                                                = var.workspace_name
+  resource_group_name                                 = var.resource_group_name
+  location                                            = var.location
+  access_connector_id                                 = var.access_connector_id
+  customer_managed_key_enabled                        = var.customer_managed_key_enabled
+  default_storage_firewall_enabled                    = var.default_storage_firewall_enabled
+  infrastructure_encryption_enabled                   = var.infrastructure_encryption_enabled
+  load_balancer_backend_address_pool_id               = var.load_balancer_backend_address_pool_id
+  sku                                                 = var.sku
+  managed_disk_cmk_key_vault_id                       = var.managed_disk_cmk_key_vault_id
+  managed_disk_cmk_key_vault_key_id                   = var.managed_disk_cmk_key_vault_key_id
+  managed_disk_cmk_rotation_to_latest_version_enabled = var.managed_disk_cmk_rotation_to_latest_version_enabled
+  managed_services_cmk_key_vault_id                   = var.managed_services_cmk_key_vault_id
+  managed_services_cmk_key_vault_key_id               = var.managed_services_cmk_key_vault_key_id
+  managed_resource_group_name                         = var.managed_resource_group_name
+  network_security_group_rules_required               = var.network_security_group_rules_required
+  public_network_access_enabled                       = var.public_network_access_enabled
+
+  custom_parameters {
+    machine_learning_workspace_id                        = var.machine_learning_workspace_id
+    nat_gateway_name                                     = var.nat_gateway_name
+    public_ip_name                                       = var.public_ip_name
+    no_public_ip                                         = var.no_public_ip
+    public_subnet_name                                   = var.public_subnet_id != null ? local.public_subnet_parsed.subnet_name : null
+    public_subnet_network_security_group_association_id  = var.public_subnet_network_security_group_association_id
+    private_subnet_name                                  = var.private_subnet_id != null ? local.private_subnet_parsed.subnet_name : null
+    private_subnet_network_security_group_association_id = var.private_subnet_network_security_group_association_id
```

Comment:
Do we know what the purpose of these `*_subnet_network_security_group_association` arguments are? I don't see them in the ARM reference for the Databricks workspace resource, and they seem to refer to the exact same value as the corresponding `*_subnet_id` arguments.